File: 2294. Partition Array Such That Maximum Difference Is K/partitionArray.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var partitionArray = function (nums, k) {
    nums.sort((a, b) => a - b)
    let min = nums[0]
    let cnt = 1;

    for (let i = 1; i < nums.length; i++) {
        if (nums[i] - min <= k) {
            continue;
        }
        cnt++;
        min = nums[i]
    }

    return cnt
};

File: 2294. Partition Array Such That Maximum Difference Is K/partitionArray.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
    public:
        int partitionArray(vector<int>& nums, int k) {
            sort(nums.begin(), nums.end());
            int min = nums[0];
            int cnt = 1;
    
            for(int i=1; i<nums.size(); i++){
                if(nums[i] - min <= k){
                    continue;
                }
                cnt++;
                min = nums[i];
            }
            return cnt;
        }
    };